### PR TITLE
2002 uta available seats

### PR DESCRIPTION
**Pull request by JosueDOrtiz**
- Changed the car bean to have a seatsAvailable entity and recreated the equal, hashcode, tostring, and added getters and setters
- Changed User bean setter and is methods for isDriver, so that the JSON key name matched the back and front end name. Had to update one controller that is sysout using the is method and some tests that were using the set method.

### CODE REVIEW

**Review by jbennatt — APPROVED** [review]
Talked with Josue about validation issues. He is working on frontend validation and I updated the backend validation so that negative values and values larger than the total number of seats aren't allowed. This is correctly _not_ updating the available seats (and returning a 400 bad request) if the available seats is set to something invalid.

**Review by jbennatt — APPROVED** [review]
I approve, this fixes the bug Josue pointed out to me (explained in the commit statement).

File: src/main/java/com/revature/beans/User.java [hunk 1]
```diff
@@ -238,10 +238,10 @@ public String getPhoneNumber() {
 	public void setPhoneNumber(String phoneNumber) {
 		this.phoneNumber = phoneNumber;
 	}
-	public boolean isDriver() {
+	public boolean isIsDriver() {
```
Comment (carolynrehm) [inline src/main/java/com/revature/beans/User.java]: Will this not work with "getIsDriver" ? I know there were some issues with the generated getter but this seems a little redundant

File: src/main/java/com/revature/controllers/UserController.java [hunk 2]
```diff
@@ -167,7 +167,7 @@ public User getUserById(@PathVariable("id")int id) {
 	@PostMapping
 	public Map<String, Set<String>> addUser(@Valid @RequestBody User user, BindingResult result) {
 		
-		System.out.println(user.isDriver());
+		System.out.println(user.isIsDriver());
```
Comment (carolynrehm) [inline src/main/java/com/revature/controllers/UserController.java]: log or remove

File: src/test/java/com/revature/services/impl/CarServiceImplTest.java [hunk 3]
```diff
@@ -71,9 +72,9 @@ public void testUpdatingCar() {
 		
 		Car expected = new Car(1, "red", 4, "Honda", "Accord", 2015, new User());
 		when(cr.save(expected)).thenReturn(expected);
-		Car actual = csi.updateCar(expected);
+//		Car actual = csi.updateCar(expected);
```
Comment (carolynrehm) [inline src/test/java/com/revature/services/impl/CarServiceImplTest.java]: remove commented code

File: src/main/java/com/revature/beans/User.java (hunk 1 above)
Comment (jbennatt) [reply src/main/java/com/revature/beans/User.java]: getIsDriver appears to work, so I changed it.

File: src/main/java/com/revature/controllers/UserController.java (hunk 2 above)
Comment (jbennatt) [reply src/main/java/com/revature/controllers/UserController.java]: removed.

File: src/test/java/com/revature/services/impl/CarServiceImplTest.java (hunk 3 above)
Comment (jbennatt) [reply src/test/java/com/revature/services/impl/CarServiceImplTest.java]: removed

**Review by carolynrehm — APPROVED** [review]

**Pull request merged**